From 57bc616fc3664ebe4be8d85bd62440116c2f9461 Mon Sep 17 00:00:00 2001
From: Jaeyoon Jung <jaeyoon.jung@lge.com>
Date: Mon, 13 Aug 2018 14:29:34 +0900
Subject: [PATCH] client: Exit on a fatal connection error

Terminate a client if there is an error such as EPIPE or ECONNRESET.
This is a safe method for a client that does not handle a fatal
connection error properly, but not recommended for the upstream as
ideally a client should be able to decide what to do on a broken
connection.

Signed-off-by: Jaeyoon Jung <jaeyoon.jung@lge.com>
---
Upstream-Status: Inappropriate [webOS specific]

 src/wayland-client.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/wayland-client.c b/src/wayland-client.c
index 9cf2793..d4fbeba 100644
--- a/src/wayland-client.c
+++ b/src/wayland-client.c
@@ -156,6 +156,13 @@ display_fatal_error(struct wl_display *display, int error)
 	display->last_error = error;
 
 	display_wakeup_threads(display);
+
+	if (error == EPIPE || error == ECONNRESET) {
+		wl_log("Exit client due to a fatal connection failure: %s\n",
+				strerror(error));
+		pthread_mutex_unlock(&display->mutex);
+		exit(1);
+	}
 }
 
 /**
